clang/lib/Tooling/DependencyScanning/IncludeTreeActionController.cpp

```diff
@@ -418,13 +418,18 @@ void IncludeTreeBuilder::enteredInclude(Preprocessor &PP, FileID FID) {
   if (!StartedEnteringIncludes) {
     StartedEnteringIncludes = true;
 
+    SmallVector<OptionalFileEntryRef> UIDToFE;
+    PP.getFileManager().GetUniqueIDMapping(UIDToFE);
```

Comment:
Is `GetUniqueIDMapping` deterministic? It looks like if there are multiple paths with the same UID you will get the one that is found first during iteration of a `StringMap`, which is unordered.

Reply:
It should be, we use it in `ASTWriter`. For the same UID, it will pick `FileEntryRef` with the most `<` name:

```c++
void FileManager::GetUniqueIDMapping(
    SmallVectorImpl<OptionalFileEntryRef> &UIDToFiles) const {

    // Add this file if it's the first one with the UID, or if its name is
    // better than the existing one.
    OptionalFileEntryRef &ExistingFE = UIDToFiles[FE.getUID()];
    if (!ExistingFE || FE.getName() < ExistingFE->getName())
      ExistingFE = FE;
}
```

Reply:
(It won't necessarily be the name by which it was included in the PCH, but that wasn't the case prior this patch either.)

Reply:
Oops, I misread this code and missed the `<` part.  Thanks!